### ISSUE TITLE
Only concrete class can be decorated with @total_ordering

### ISSUE THREAD

**Issue by mthuurne** [python/mypy#8539]
When I check this code with mypy:

```py
from abc import abstractmethod
from functools import total_ordering

@total_ordering
class PriorityMixin:

    @abstractmethod
    def getName(self) -> str: ...

    @abstractmethod
    def getPriority(self) -> int: ...

    def __hash__(self) -> int:
        return hash(self.getName())

    def __eq__(self, other: object) -> bool:
        if isinstance(other, PriorityMixin):
            return self.getName() == other.getName()
        else:
            return NotImplemented

    def __lt__(self, other: object) -> bool:
        if isinstance(other, PriorityMixin):
            prioCmp = self.getPriority() - other.getPriority()
            if prioCmp == 0:
                return self.getName() < other.getName()
            else:
                return prioCmp < 0
        else:
            return NotImplemented
```

It reports:

```
testcase.py:4: error: Only concrete class can be given where "Type[PriorityMixin]" is expected  [misc]
    @total_ordering
     ^
```

I'm using mypy 0.770 under Python 3.7.3 on openSUSE Linux.

I don't see a reason why an abstract class cannot be decorated with `@total_ordering`.

In [typeshed commit e404e15](https://github.com/python/typeshed/commit/e404e1592d8e1de9099c2dabc9fdb0e55c2d2070) the signature of `total_ordering` was changed to preserve the original type. I think that change makes sense until full `total_ordering` support is implemented (#4610), but it does trigger the issue described above.

I think this issue may be a variation of #5374, since that describes a similar problem when using `@dataclass` as an annotation.

**Comment by JukkaL** [comment]
Yeah, this looks like a bug. The best fix might be to generally relax the requirement to have a concrete class as a type object.

**Comment by jakob-keller** [comment]
I am experiencing the same issue with mypy 0.770, whereas this appears to pass in 0.761. Maybe the root cause may be narrowed down to a regression introduced in 0.770.

**Comment by mthuurne** [comment]
> I am experiencing the same issue with mypy 0.770, whereas this appears to pass in 0.761. Maybe the root cause may be narrowed down to a regression introduced in 0.770.

That's because of the new typeshed that's included with mypy 0.770. The change in typeshed is not the root cause though, it only exposed the existing limitation in mypy.

**Comment by erictraut** [comment]
This bug appears to have been fixed at some point. I'm not able to repro it with the latest version of mypy.
